### PR TITLE
Feature/common thymeleaf fragments

### PR DESCRIPTION
**Pull request by Joseph-Nash**
Created a folder of common Thymeleaf fragments that can be used across projects via a maven dependency to avoid unnecessary replication and keep track of the latest changes. Added two variables that need to be set in the user's project (in the header and piwik) as they are specific to the service.

__CPS_164__

### Type of change

* [] Bug fix
* [x] New feature
* [] Breaking change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: src/main/resources/templates/error.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      layout:decorate="~{layouts/baseLayout}">
+
+    <head>
+        <title>Server error</title>
+    </head>
+    <body>
+        <div layout:fragment="content">
+
+            <div class="text">
+                <h1 class="heading-xlarge">Sorry we are experiencing technical difficulties</h1>
```
Comment (BSaunders95) [inline src/main/resources/templates/error.html]: Can the class be updated here to `govuk-heading-l` for use with the most up to date stylesheet please

File: src/main/resources/templates/fragments/header.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                <div class="govuk-header__container govuk-width-container">
+                    <div class="header-global">
+                        <div class="govuk-header__logo">
+                            <a href="/" title="Go to the companies house homepage" id="logo" class="govuk-header__link govuk-header__link--homepage">
+                                <span class="govuk-header__logotype-text"> Companies House </span>
+                            </a>
+                        </div>
+                    </div>
+
+                     <div class="govuk-header__content">
```
Comment (BSaunders95) [inline src/main/resources/templates/fragments/header.html]: Can this be changed so the div only appears if `headerText` is defined

**Comment by BSaunders95** [comment]
We'll need the fragments to be well documented in the README. Will this be done in a separate PR?

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>uk.gov.companieshouse</groupId>
+	<artifactId>common-web-java</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
Comment (justinblandford1) [inline pom.xml]: Need to check the process around release versions.    @BSaunders95  If this review is now complete, do we manually set this version to 1.0.0-rc1 or is there a Maven release plugin that does this for us automatically via a Jenkins job or something?

Comment (justinblandford1) [reply pom.xml]: Spoke with Bradley, the pom version is irrelevant, the versioning is done via tagging so no issue here - just set it to unversioned

**Pull request merged**